### ISSUE TITLE
clang-format QualifierOrder produces wrong orders

### ISSUE THREAD

**Issue by lizhuimu** [llvm/llvm-project#58010]
```console
$ clang-format-16 --version
Ubuntu clang-format version 16.0.0-++20220925052850+bcb1397bda66-1~exp1~20220925172953.390
```

`.clang-format`:
```yaml
---
BasedOnStyle: google
QualifierAlignment: Custom
QualifierOrder: ['inline', 'constexpr', 'static', 'type', 'const', 'volatile']
```

input:
```cpp
template <typename T1>
std::array<char const*, 2> const Foo<T1>::kBar {"a", "b"};
```

bad output:
```cpp
template <typename T1>
std::array<char const*, 2> Foo<T1>::kBar const {"a", "b"};
```

**Comment by sean-mcmanus** [comment]
@EugeneZelenko This also repros with the very simple case of ` std::string const s;` -- let me know if I should open a new issue for that. It works when the "std::" is removed via "using namespace std;".

**Comment by EugeneZelenko** [comment]
@sean-mcmanus: It's better to ask Clang-format developers.

**Comment by rymiel** [comment]
This should be fixed by cd7ab4b5c1684dcc60de027700177adfa096b98c
